auth/middleware.go

```diff
@@ -87,37 +95,53 @@ func NewMiddleware(oAuthConfig OAuthConfig, options Options) *Middleware {
 	return m
 }
 
-// Authenticate authenticates a request and returns the Token if validation was successful, otherwise error is returned
-func (m *Middleware) Authenticate(r *http.Request) (Token, error) {
+// authenticate authenticates a request and returns the Token and the client certificate if validation was successful,
+// otherwise error is returned
+func (m *Middleware) authenticate(r *http.Request) (Token, *x509.Certificate, error) {
 	// get Token from Header
 	rawToken, err := extractRawToken(r)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	token, err := m.parseAndValidateJWT(rawToken)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	const forwardedClientCertHeader = "x-forwarded-client-cert"
```

Comment:
Starting with the prefix `x-` I assume this header name is not an industry standard? Maybe it makes sense to make this configurable when instantiating the middleware? At the same time we could add configuration to make the `Authorization` header configurable (though this is a standard header).

Reply:
Good idea, but i guess we shall handle that in a separate PR. And yes, it is no industry standard, but at least documented in the CF docs: https://docs.cloudfoundry.org/concepts/http-routing.html#forward-client-cert